config/v3_1_experimental/types/ignition.go

```diff
@@ -15,54 +15,23 @@
 package types
 
 import (
-	"net/url"
-
 	"github.com/coreos/go-semver/semver"
 
 	"github.com/coreos/ignition/v2/config/shared/errors"
-	"github.com/coreos/ignition/v2/config/util"
 
 	"github.com/coreos/vcontext/path"
 	"github.com/coreos/vcontext/report"
 )
 
-func (c ConfigReference) Key() string {
-	if c.Source == nil {
-		return ""
-	}
-	return *c.Source
-}
-
-func (cr ConfigReference) Validate(c path.ContextPath) (r report.Report) {
-	r.AddOnError(c.Append("source"), validateURLNilOK(cr.Source))
-	r.AddOnError(c.Append("httpHeaders"), cr.validateSchemeForHTTPHeaders())
-	return
+func (v Ignition) Semver() (*semver.Version, error) {
+	return semver.NewVersion(v.Version)
 }
 
-func (cr ConfigReference) validateSchemeForHTTPHeaders() error {
-	if len(cr.HTTPHeaders) < 1 {
-		return nil
+func (ic IgnitionConfig) Validate(c path.ContextPath) (r report.Report) {
+	for i, res := range ic.Merge {
+		r.AddOnError(c.Append("merge", i), res.validateRequiredSource())
```

Comment:
Nope.  Since the `Replace` member embeds a `Resource` struct rather than pointing to it, the absence of `Source` is what distinguishes whether `Replace` was specified at all.

Reply:
Heh, right yeah. (This is actually part of what informed my comment in https://github.com/coreos/fcct/pull/90#issuecomment-610041275 -- it'd solve the issue here nicely).